prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -188,6 +190,49 @@ class DeliveryFunction(
         return this.getDetailedView(request, id)
     }
 
+    /**
+     *Endpoint for intermediary receivers to verify status of messages. It passes
+     * a null engine to the retrieveMetadata function because Azure gets upset if there
+     * are any non-annotated parameters in the method signature other than ExecutionContext
+     * and we needed the engine to be a parameter so it can be mocked for tests
+     *
+     */
+    @FunctionName("getEtorMetadataForDelivery")
+    fun getEtorMetadata(
+        @HttpTrigger(
+            name = "getEtorMetadataForDelivery",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/{reportId}/delivery/etorMetadata"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("reportId") reportId: UUID,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        return this.retrieveETORIntermediaryMetadata(request, reportId, context, null)
+    }
+
+    /**
+     * Function for finding the associated report ID that the intermediary knows about given the report ID that the receiver is
+     * given from report stream
+     */
+    override fun getLookupId(reportId: UUID): UUID? {
```

Comment:
Can you please explain the implementation of this method? In your use-case, there should only ever be one root report because we will not be batching messages to receivers, correct? IF we are batching messages to receivers, then there will be multiple root reports, but I fail to see how the oldest one would necessarily be the correct one?

This comment is based on my understanding that `getRootReports` returns reports that do not have a parent (If receivers have batching turned off this will never be the case because each report containing a single message can only ultimately come from a single submitted report). Based on the logic I see here, I think your understanding is that it returns ALL ancestors of a report?

That said, this logic could simply be `return reportGraph.getRootReport(reportId)` because getRootReport will throw tooManyRows exception if there is more than one result. Then, I recommend updating `retrieveETORIntermediaryMetadata` to take in the single line logic as a function param instead of the current `override fun getLookupId` pattern.

Reply:
BONUS: Notice the description of getRootReport:
```
     * Recursively goes up the report_linage table from any report until it reaches
     * a report with an action type of "receive" (the root report)
     *
     * This will return null if no report with action type "receive" is present or if
     * the root is passed in
```

Reply:
Yes, you are correct.  We technically don't support batching as of this moment and would require additional work on our end to support in the future.  Given that, we have migrated to using `getRootReport` (singular).

Reply:
I also have an [outstanding question to you in Slack](https://flexion.slack.com/archives/C048S3GPSQG/p1714510584647929?thread_ts=1714503553.513999&cid=C048S3GPSQG) about changing the pattern for `getLookupId`.